api/pkg/transformer/jsonpath/jsonpath_test.go

```diff
@@ -182,6 +187,98 @@ func TestCompiled_Lookup(t *testing.T) {
 	}
 }
 
+func TestCompiledWithOption_LookupFromContainer(t *testing.T) {
+	var rawRequestData types.JSONObject
+	var modelResponseData types.JSONObject
+
+	json.Unmarshal([]byte(rawRequestJson), &rawRequestData)
+	json.Unmarshal([]byte(modelResponseJson), &modelResponseData)
+
+	jsonContainer := types.JSONObjectContainer{
+		spec.JsonType_RAW_REQUEST:    rawRequestData,
+		spec.JsonType_MODEL_RESPONSE: modelResponseData,
+	}
+	testCases := []struct {
+		desc        string
+		opt         JsonPathOption
+		sourceJSONs types.JSONObjectContainer
+		want        interface{}
+		wantErr     bool
+		err         error
+	}{
+		{
+			desc: "without default value",
+			opt: JsonPathOption{
+				JsonPath: "$.signature_name",
+			},
+			sourceJSONs: jsonContainer,
+			want:        "predict",
+			wantErr:     false,
+			err:         nil,
+		},
+		{
+			desc: "not exist key without default value",
+			opt: JsonPathOption{
+				JsonPath: "$.not_exist_key",
+			},
+			sourceJSONs: jsonContainer,
+			want:        nil,
+			wantErr:     false,
+			err:         nil,
+		},
+		{
+			desc: "not exist key with default value",
+			opt: JsonPathOption{
+				JsonPath:     "$.not_exist_key",
+				DefaultValue: "4.4",
+				TargetType:   spec.ValueType_FLOAT,
+			},
+			sourceJSONs: jsonContainer,
+			want:        4.4,
+			wantErr:     false,
+			err:         nil,
+		},
+		{
+			desc: "null array with default value",
+			opt: JsonPathOption{
+				JsonPath:     "$.null_array",
```

Comment:
Do we know if it's an array? It's just empty json field right?

Reply:
oh yeah, it is not necessary an array. This is the case that we face when user specified their array value `null` instead of `[]`, also it can be applied to arbitary json field. I'll update the desc